### PR TITLE
Implements Soluti Oauth User Discovery

### PR DESCRIPTION
**Pull request by joaosalless**
Discovery user before authenticating to find out which cloud the user exists.

### CODE REVIEW

File: src/Memed/Soluti/Auth/Session.php [hunk 1]
```diff
@@ -103,6 +82,90 @@ public function applicationToken(Credentials $credentials, string $cloud): Appli
...
+     *
+     * @param  Cloud  $cloud
+     * @param  Credentials  $credentials
+     * @return DiscoveredOauthUser
+     */
+    private function oauthUserDiscoveryRequest(
+        Cloud $cloud,
+        Credentials $credentials
+    ): DiscoveredOauthUser
+    {
```
Comment (DavidCardoso) [inline src/Memed/Soluti/Auth/Session.php]: **nip**:  Following [PSR-12](https://www.php-fig.org/psr/psr-12/)

```php
private function oauthUserDiscoveryRequest(
        Cloud $cloud,
        Credentials $credentials
): DiscoveredOauthUser {
    // code here...
}
```

File: tests/Memed/Soluti/Auth/SessionTest.php [hunk 2]
```diff
@@ -47,15 +47,40 @@ protected function setUp(): void
         $this->applicationToken = new ApplicationToken(
             'some-token',
             'some-type',
-            'VAULT_ID'
+            'BIRD_ID'
         );
+
+        $this->cloudMock = m::mock(Cloud::class, [
+            CloudAuthentication::CLOUD_NAME_BIRD_ID,
+            $this->birdidUrl,
```
Comment (DavidCardoso) [inline tests/Memed/Soluti/Auth/SessionTest.php]: **typo**: `$this->birdIdUrl`

File: tests/Memed/Soluti/SignerTest.php [hunk 3]
```diff
@@ -9,6 +9,7 @@
 use Memed\Soluti\Auth\CloudAuthentication;
 use Memed\Soluti\Auth\Credentials;
 use Memed\Soluti\Auth\Session;
+use Memed\Soluti\Auth\Token;
 use Memed\Soluti\Auth\Token as AuthToken;
```
Comment (DavidCardoso) [inline tests/Memed/Soluti/SignerTest.php:12]: Remove this unnecessary/duplicated class. Choose between `Token` or `AuthToken`

**Review by DavidCardoso — CHANGES_REQUESTED** [review]

File: src/Memed/Soluti/Auth/Session.php (hunk 1 above)
Comment (joaosalless) [reply src/Memed/Soluti/Auth/Session.php]: Done.

File: tests/Memed/Soluti/Auth/SessionTest.php (hunk 2 above)
Comment (joaosalless) [reply tests/Memed/Soluti/Auth/SessionTest.php]: Done.

File: tests/Memed/Soluti/SignerTest.php (hunk 3 above)
Comment (joaosalless) [reply tests/Memed/Soluti/SignerTest.php:12]: Done.

**Review by DavidCardoso — APPROVED** [review]

**Pull request merged**